graylog2-server/src/main/java/org/graylog2/shared/utilities/ContainerRuntimeDetection.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.shared.utilities;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.Callable;
+import java.util.stream.Stream;
+
+public class ContainerRuntimeDetection {
+
+    public static Boolean isRunningInsideContainer() {
+        return isRunningInsideContainer(
+                () -> Files.lines(Paths.get("/proc/self/cgroup")),
+                () -> Files.lines(Paths.get("/proc/self/mountinfo")),
+                Path.of("/.dockerenv")
```

Comment:
Today I learned that at least podman and cri-o add a similar file: `/run/.containerenv`. See chttps://github.com/cri-o/cri-o/issues/5461.

Not sure if that would be a detection worth adding, as we have the generic cgroup-based detection code below. But wanted to mention it nevertheless.

Reply:
Added in 9557a963c9.